### PR TITLE
CORE-6146 - add configuration for websocket idle timeout

### PR DESCRIPTION
**Pull request by conal-smith**
https://github.com/corda/corda-runtime-os/pull/2002

### CODE REVIEW

File: data/config-schema/src/main/resources/net/corda/schema/configuration/rpc/1.0/corda.rpc.json [hunk 1]
```diff
@@ -48,6 +48,20 @@
...
+    },
+    "websocket": {
+      "description": "Configuration settings for WebSockets.",
+      "type": "object",
+      "default": {},
+      "properties": {
+        "idleTimeoutMs": {
+          "description": "Amount of time in milliseconds for a websocket connection to be idle before timing out and closing the connection.",
+          "type": "integer",
+          "minimum": 0,
```
Comment (vkolomeyko) [inline data/config-schema/src/main/resources/net/corda/schema/configuration/rpc/1.0/corda.rpc.json:60]: Shall put something like:
```suggestion
          "minimum": 100,
```
or else value of `0` may be interpreted as "never".

File: data/config-schema/src/main/resources/net/corda/schema/configuration/rpc/1.0/corda.rpc.json [hunk 2]
```diff
@@ -48,6 +48,20 @@
...
         }
       }
+    },
+    "websocket": {
+      "description": "Configuration settings for WebSockets.",
+      "type": "object",
+      "default": {},
+      "properties": {
+        "idleTimeoutMs": {
+          "description": "Amount of time in milliseconds for a websocket connection to be idle before timing out and closing the connection.",
```
Comment (vkolomeyko) [inline data/config-schema/src/main/resources/net/corda/schema/configuration/rpc/1.0/corda.rpc.json]: Value of `0` will be interpreted as "never". From `org.eclipse.jetty.io.EndPoint#setIdleTimeout`: `Params:
idleTimeout – the idle timeout in MS. Timeout <= 0 implies an infinite timeout`.

Therefore:
```suggestion
          "description": "Amount of time in milliseconds for a websocket connection to be idle (i.e. no messages are either sent or received) before timing out and closing the connection. Value of 0 means such idle check will be disabled.",
```

File: gradle.properties [hunk 3]
```diff
@@ -7,7 +7,7 @@ cordaProductVersion = 5.0.0
 # NOTE: update this each time this module contains a breaking change
 ## NOTE: currently this is a top level revision, so all API versions will line up, but this could be moved to
 ##   a per module property in which case module versions can change independently.
-cordaApiRevision = 172
+cordaApiRevision = 173
```
Comment (vkolomeyko) [inline gradle.properties]: The version has moved on by now.

File: data/config-schema/src/main/resources/net/corda/schema/configuration/rpc/1.0/corda.rpc.json (hunk 2 above)
Comment (sprintengine) [reply data/config-schema/src/main/resources/net/corda/schema/configuration/rpc/1.0/corda.rpc.json]: Well spotted, thanks!

**Review by vkolomeyko — DISMISSED** [review]
LGTM

**Review by vkolomeyko — APPROVED** [review]
LGTM

**Pull request merged**